src/metatensor/models/utils/device.py

```diff
@@ -0,0 +1,115 @@
+import warnings
+from typing import List
+
+import torch
+
+
+def get_available_devices() -> List[torch.device]:
+    """Returns a list of available torch devices.
+
+    This function returns a list of available torch devices, which can
+    be used to specify the devices on which to run a model.
+
+    :return: The list of available torch devices.
+    """
+    devices = [torch.device("cpu")]
+    if torch.cuda.is_available():
+        device_count = torch.cuda.device_count()
+        for i in range(device_count):
+            devices.append(torch.device(f"cuda:{i}"))
+    if torch.backends.mps.is_built() and torch.backends.mps.is_available():
+        devices.append(torch.device("mps"))
+    return devices
+
+
+def pick_devices(
+    requested_device: str,
+    available_devices: List[torch.device],
+    architecture_devices: List[str],
+) -> List[torch.device]:
+    """Picks the devices to use for training.
+
+    This function picks the devices to use for training based on the
+    requested device, the available devices, and the list of devices
+    supported by the architecture.
+
+    The choice is based on the following logic.
```

Comment:
is this missing a description of the logic?

Reply:
Sorry, it's now there